File: JavaScript Problems/addWithout.js
// Given two integers a and b, return the sum of the two integers without using the operators + and -.

function getSum(a, b) {
    while (b !== 0) {
        let carry = a & b;  // Calculate carry
        a = a ^ b;          // Add without carrying
        b = carry << 1;     // Shift carry to the left
    }
    return a;
}

File: Python Problems/addWithout.py
# Given two integers a and b, return the sum of the two integers without using the operators + and -.

def get_sum(a, b):
    mask = 0xFFFFFFFF
    while b:
        a, b = (a ^ b) & mask, ((a & b) << 1) & mask
    return a if a <= 0x7FFFFFFF else ~(a ^ mask)